### PR TITLE
[INLONG-9862][Manager] Support submit flink job for offline sync

### LINKED ISSUE #9862: [Feature][Manager] Support submit flink job for offline data sync
### Description

Support submit flink job for offline data sync

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by aloyszhang**
- Fixes #9862 
### Background

For real-time data sync, the whole procedure is as follows:

image

Currently, InLong does not support submitting Flink batch jobs for offline data sync.

The main differences between offline sync and real-time sync are as follows:
- Offline synchronization requires corresponding scheduling information to be configured
- Offline task submission is triggered  by the scheduling system instead of manager

The interaction process with the scheduling system is as follows:
- Register the scheduling information at the point of approval 
- The scheduling system generates an instance to submit Flink batch tasks by InLong callback.

Design principles for the association between synchronization tasks and the scheduling system are as follows:

- Management of task configuration and scheduling configuration are independent of each other.
- The granularity of scheduling information configuration is at the stream level: there are multiple streams under a group, fine-grained scheduling control should be at the stream level, and the scheduling information registered with the scheduling system should include groupID and streamID.
- Whether to register scheduling information with the scheduling system depends on the approval status of the task: If the task is in an approved state: save the scheduling information and register it with the scheduling system, otherwise
, only save the scheduling information 

Furthermore, it is important to note the distinction between configuring streams and scheduling. 
- In the configuration process, Flink tasks are not submitted,
- In the scheduling process, Flink tasks are submitted.

This is identified by the status of the stream.

PS: Scheduling management is not covered by this pull request.

### Motivation

Support submit flink job for offline sync

### Modifications

Ensure the Flink batch task will be submitted only if the stream is in the CONFIG_SUCCESSFUL state.

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: inlong-manager/manager-plugins/base/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java [hunk 1]
```diff
@@ -98,6 +99,14 @@ public ListenerResult listen(WorkflowContext context) throws Exception {
         }
 
         for (InlongStreamInfo streamInfo : streamInfos) {
+            boolean isRealTimeSync = InlongConstants.DATASYNC_REALTIME_MODE
+                    .equals(groupResourceForm.getGroupInfo().getInlongGroupMode());
+            // do not submit flink job if the group mode is offline and the stream is not config successfully
+            if (isRealTimeSync && !StreamStatus.CONFIG_SUCCESSFUL.getCode().equals(streamInfo.getStatus())) {
```
Comment (fuweng11) [inline inlong-manager/manager-plugins/base/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java:105]: `streamInfo.getStatus()` is always `StreamStatus.CONFIG_ING`.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/listener/sort/SortConfigListener.java [hunk 2]
```diff
@@ -125,6 +126,13 @@ public ListenerResult listen(WorkflowContext context) throws WorkflowListenerExc
 
         try {
             for (InlongStreamInfo streamInfo : streamInfos) {
+                // do not build sort config if the group mode is offline and the stream is not config successfully
+                if (InlongConstants.DATASYNC_OFFLINE_MODE.equals(groupInfo.getInlongGroupMode())
+                        && !StreamStatus.CONFIG_SUCCESSFUL.getCode().equals(streamInfo.getStatus())) {
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/listener/sort/SortConfigListener.java:131]: Ditto.

Comment (aloyszhang) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/listener/sort/SortConfigListener.java]: fixed, PTAL

File: inlong-manager/manager-plugins/base/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java (hunk 1 above)
Comment (aloyszhang) [reply inlong-manager/manager-plugins/base/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java]: fixed, PTAL

File: inlong-manager/manager-plugins/base/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java [hunk 3]
```diff
@@ -131,30 +140,24 @@ public ListenerResult listen(WorkflowContext context) throws Exception {
...
                 flinkOperation.genPath(flinkInfo, dataflow);
-                // only start job for real-time mode
-                if (isRealTimeSync) {
-                    flinkOperation.start(flinkInfo);
-                    log.info("job submit success for groupId = {}, streamId = {}, jobId = {}", groupId,
-                            streamInfo.getInlongStreamId(), flinkInfo.getJobId());
-                }
+                flinkOperation.start(flinkInfo);
+                log.info("job submit success for groupId = {}, streamId = {}, jobId = {}", groupId,
+                        streamInfo.getInlongStreamId(), flinkInfo.getJobId());
```
Comment (baomingyu) [inline inlong-manager/manager-plugins/base/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java:160]: It is recommended to add real-time/offline identification information in the log content

Comment (aloyszhang) [reply inlong-manager/manager-plugins/base/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java:160]: Make sense. Fixed. PTAL again.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/InlongConstants.java [hunk 4]
```diff
@@ -141,6 +141,8 @@ public class InlongConstants {
      */
     public static final String DATAFLOW = "dataflow";
 
+    public static final String STREAM_CONFIG_STATUS = "stream.config.status";
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/InlongConstants.java:144]: Please rename, this information is only used to determine whether scheduling is enabled, and the naming of this parameter does not match its actual meaning

File: inlong-manager/manager-plugins/base/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java [hunk 5]
```diff
@@ -98,6 +99,14 @@ public ListenerResult listen(WorkflowContext context) throws Exception {
         }
 
         for (InlongStreamInfo streamInfo : streamInfos) {
+            boolean isOfflineSync = InlongConstants.DATASYNC_OFFLINE_MODE
+                    .equals(groupResourceForm.getGroupInfo().getInlongGroupMode());
+            // do not submit flink job if the group mode is offline and the stream is not config successfully
+            if (isOfflineSync && !StreamParseUtils.isStreamConfigSuccess(streamInfo)) {
```
Comment (fuweng11) [inline inlong-manager/manager-plugins/base/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java:105]: Ditto.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/StreamParseUtils.java [hunk 6]
```diff
@@ -154,4 +159,22 @@ public static StreamPipeline parseStreamPipeline(String tempView, String inlongS
...
+        if (StringUtils.isEmpty(key)
+                || streamInfo == null || streamInfo.getExtList() == null || streamInfo.getExtList().isEmpty()) {
+            return null;
+        }
+        for (InlongStreamExtInfo ext : streamInfo.getExtList()) {
+            if (key.equalsIgnoreCase(ext.getKeyName())) {
+                return ext.getKeyValue();
+            }
+        }
+        return null;
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/StreamParseUtils.java:172]:
```suggestion
        if (StringUtils.isNotBlank(key)
                && streamInfo != null && CollectionUtils.isNotEmpty(streamInfo.getExtList())) {
                  for (InlongStreamExtInfo ext : streamInfo.getExtList()) {
            if (key.equalsIgnoreCase(ext.getKeyName())) {
                return ext.getKeyValue();
            }
        }
       
        return null;
```

**Comment by aloyszhang** [comment]
Update:

After discussing with @fuweng11, we will add a ServiceTask named Init-Schedule at the endpoint of StreamResourceProcessForm
image

**Review by fuweng11 — APPROVED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/InlongConstants.java [hunk 7]
```diff
@@ -141,6 +141,10 @@ public class InlongConstants {
      */
     public static final String DATAFLOW = "dataflow";
 
+    public static final String REGISTER_SCHEDULE_SUCCESS = "register.schedule.success";
+
+    public static final String TRUE = "true";
```
Comment (dockerzhang) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/InlongConstants.java:146]: `REGISTERED` may be better?

Comment (aloyszhang) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/InlongConstants.java]: Fixed. PTAL

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**